.github/workflows/release.yml

```diff
@@ -43,7 +43,7 @@ jobs:
       - name: Install dependencies
         run: npm ci
       - name: Version, release and publish packages
-        run: npx nx release --yes --dry-run
+        run: npx nx release --yes --dry-run -- --skip-nx-cache
```

Comment:
Why do we need to skip the cache here?

Reply:
Apparently, the package.json file changes (when bumping the version) are not detected and the `build` target (that `nx-release-publish` depends on) is using the cache output. Meaning 
1. the package.json in the `dist` folder is using the older version
2. there is nothing to publish 

So I configured the `nx-release-publish` target to forward arguments to `build`.
An alternative approach: change or create a new `namedInput` that explicitly includes the projects package.json. How about that?

Reply:
The alternative approach seems more explicit and a tiny bit more self documenting.
Do you think it pays off t write something about build and release i the `CONTRIBUTING.md`?

Let me know what you think.

Reply:
Yes, i agree it's better to configure inputs correctly. I looked at the [nx.json](https://github.com/code-pushup/cli/blob/main/nx.json) and it seems like the `build` target and the related executor should catch changes from package.json... I will try this further...

Reply:
The release config and process can be a bit confusing, so yes, it's worth mentioning some bits of how it works and how it is configured. I will also add some external references to avoid repeating common knowledge...
I'll take care of it in this PR

Reply:
Thanks a LOT for the extra effort!

Reply:
After testing it seems like the build cache is behaving correctly, I adapted the nx release config and arguments accordingly.